tests/test_custom_models.py

```diff
@@ -2127,13 +2185,26 @@ def test_multiple_active_adapters_merge_and_unmerge(
     @parameterized.expand(MULTIPLE_ACTIVE_ADAPTERS_TEST_CASES)
     def test_merge_layers_multi(self, test_name, tuner_method, config_cls, config_kwargs_1, config_kwargs_2):
         torch.manual_seed(0)
-        model = MLP(bias=tuner_method != "ia3").to(self.torch_device).eval()
+
+        model = self.resolve_model_cls(tuner_method)
+        model = model.to(self.torch_device).eval()
 
         config_1 = config_cls(**config_kwargs_1)
         config_2 = config_cls(**config_kwargs_2)
 
         model = get_peft_model(model, config_1)
 
+        # the assumption that the output of the combined output of two adapters is != to the output of one
+        # adapter is not true for unmodified trainable tokens as they just mimic the existing embedding matrix.
+        # therefore, we modify the weights so that the adapter weights differs from the embedding weights. in this
+        # case we even use 20*rand to be very distinct to adapter 2 since we're comparing outputs and not embeddings
+        # with rather high tolerance values. this is also the reason why `init_weights` is not sufficient here and
+        # when using `<peft method>.trainable_token_indices` we do not have the utility of `init_weights` anyway.
+        if "trainable_tokens" in tuner_method:
+            model.emb.token_adapter.trainable_tokens_delta["default"].data = 20 * torch.rand_like(
```

Comment:
So I tried out this test again and it passed for me with `2 *` instead of `20 *`. Not sure what is different than last time, but let's go with `2`? Test on CPU and GPU.

Reply:
If it was flaky before it will probably be flaky some point later. I don't see why we couldn't continue with 20. It is rather arbitrary.